mmr/helper.go

```diff
@@ -100,20 +111,15 @@ func LeafIndexToMMRSize(index uint64) uint64 {
 	return 2*leavesCount - uint64(peakCount)
 }
 
-func pop(ph [][]byte) ([]byte, [][]byte) {
-	if len(ph) == 0 {
-		return nil, ph[:]
+// pop removes the last item from a slice and returns it
+func pop(slice *[][]byte) []byte {
+	var sliceCopy = *slice
+	if len(sliceCopy) == 0 {
```

Comment:
should this be allowed? 
Since you're returning `nil` in case the length is zero, the caller has to check for it. Better have the caller check before calling this function and avoiding this case at all?